clients/hadoopfs/src/test/java/io/lakefs/auth/GetCallerIdentityV4PresignerTest.java

```diff
@@ -0,0 +1,66 @@
+package io.lakefs.auth;
+
+import com.amazonaws.Request;
+import com.amazonaws.auth.AWSSessionCredentials;
+import io.lakefs.Constants;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public class GetCallerIdentityV4PresignerTest {
+    public static AWSSessionCredentials newMockAWSCreds() {
+        return new AWSSessionCredentials() {
+            @Override
+            public String getSessionToken() {
+                return "sessionToken";
+            }
+
+            @Override
+            public String getAWSAccessKeyId() {
+                return "accessKeyId";
+            }
+
+            @Override
+            public String getAWSSecretKey() {
+                return "secretKey";
+            }
+        };
+    }
+
+    @Test
+    public void testPresignAsURL() throws Exception {
+        AWSSessionCredentials awsCreds = GetCallerIdentityV4PresignerTest.newMockAWSCreds();
+        GetCallerIdentityV4Presigner stsPresigner = new GetCallerIdentityV4Presigner();
+        GeneratePresignGetCallerIdentityRequest stsReq = new GeneratePresignGetCallerIdentityRequest(
+                new URI("https://sts.amazonaws.com"),
+                awsCreds,
+                new HashMap<String,String>(){{
+                    put(Constants.DEFAULT_AUTH_PROVIDER_SERVER_ID_HEADER, "lakefs-host");
+                }},
+                60
+        );
+        Request<GeneratePresignGetCallerIdentityRequest> req = stsPresigner.presignRequest(stsReq);
+        URL url = STSGetCallerIdentityPresigner.convertRequestToUrl(req);
+
+        String pattern = "https://sts\\.amazonaws\\.com/\\?"+
+                "X-Amz-Date=\\d{8}T\\d{6}Z&"
+                +"Action=GetCallerIdentity&"
+                + "X-Amz-Algorithm=AWS4-HMAC-SHA256&"+
+                "X-Amz-Signature=[a-f0-9]{64}&"+
+                "Version=2011-06-15&"+
+                "X-Amz-SignedHeaders=host%3Bx-lakefs-server-id&"+
+                "X-Amz-Security-Token=[^&]+&"+
+                "X-Amz-Credential="+ awsCreds.getAWSAccessKeyId()+"%2F\\d{8}%2Fus-east-1%2Fsts%2Faws4_request&"+
+                "X-Amz-Expires=\\d+";
+
+        Pattern compiledPattern = Pattern.compile(pattern);
+        Matcher matcher = compiledPattern.matcher(url.toString());
+        Assert.assertEquals(String.format("URL %s does not match \npattern: %s",url, pattern), true, matcher.matches());
```

Comment:
This unit test verifies that the code returns all query parameters in a particular order.  But I do not think that this should be part of the requirement for what we are testing.

So I would really prefer to _parse_ the HRL, and then verify that the fields are correct.

Reply:
updated the test to check and compare each parameter one by one as suggested, thanks!